app/src/main/java/com/example/countrylist/base/views/BaseView.kt

```diff
@@ -0,0 +1,8 @@
+package com.example.countrylist.base.views
+
+interface BaseView<T> {
+    fun setPresenter(presenter: T)
```

Comment:
We do not need setPresenter here as we create an instance in view.
I do not think that we need generic type here.